### PR TITLE
[Task]: Improve redirects.

### PR DESCRIPTION
**Pull request by martineiber**

### CODE REVIEW

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js [hunk 1]
```diff
@@ -18,6 +18,22 @@ pimcore.helpers.sanitizeEmail = function (email) {
     return email.replace(/[^a-zA-Z0-9_\-@.+]/g,'');
 };
 
+
+// Note: Only run this once. And don't run it on already encoded strings e.g. &amp; becomes &amp;amp;
+pimcore.helpers.escapeHtml = function (html) {
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: using `Ext.util.Format.htmlEncode()`  should do the trick as well.

File: bundles/AdminBundle/Resources/public/js/pimcore/settings/redirects.js [hunk 2]
```diff
@@ -286,7 +286,15 @@ pimcore.settings.redirects = Class.create({
                 flex: 150, sortable:true, dataIndex: "expiry",
                 editor: {
                     xtype: 'datefield',
-                    format: 'Y-m-d'
+                    format: 'Y-m-d',
+                    listeners: {
+                        change: function (column, value, oldValue, eOpts) {
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/settings/redirects.js]: Maybe there's something provided by the `DateField` that covers that as well? 
https://docs.sencha.com/extjs/7.0.0/classic/Ext.form.field.Date.html

**Review by brusch — CHANGES_REQUESTED** [review]

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js (hunk 1 above)
Comment (martineiber) [reply bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: Thanks for the hint. `Ext.String.hasHtmlCharacters` is even more useful in this case to check if Html characters are present.

File: bundles/AdminBundle/Resources/public/js/pimcore/settings/redirects.js (hunk 2 above)
Comment (martineiber) [reply bundles/AdminBundle/Resources/public/js/pimcore/settings/redirects.js]: With the reset method we can restore the old valid value.

**Pull request merged**